src/components/input/Button.tsx

```diff
@@ -96,8 +103,11 @@ export default function Button({
       {...htmlAttributes}
       className={classnames(
         {
-          'focus-visible-ring transition-colors whitespace-nowrap flex items-center':
-            styled,
+          'transition-colors whitespace-nowrap flex items-center': styled,
+        },
+        styled && {
+          'focus-visible-ring': focusRing === 'focus-visible',
+          'focus:ring outline-none': focusRing === 'focus',
```

Comment:
I don't love having to add a new property to the `Button`, but it's the only way to not affect other use cases.

Basically, the custom `focus-visible-ring` aims to style focused elements based on `:focus-visible`, instead of `:focus`.

Those two are equivalent for editable elements (like inputs or elements with `contenteditable`), but not for `buttons`.

A clicked button becomes focused, but does not become `focus-visible`, preventing the focus ring to render (See https://developer.mozilla.org/en-US/docs/Web/CSS/:focus-visible for details and examples).

This effectively means that, if the `SelectNext` toggle is clicked, the listbox is opened, the first option is focused, but it does not render a focus ring, causing the second option to render a focus ring when pressing <kbd>ArrowDown</kbd> for the first time.

By using tailwind's standard `focus:ring`, we apply the ring to `:focus`, making sure the first option renders a focus ring when the listbox is opened via mouse, and then we can interact with the list via arrow keys with an intuitive behavior.

Reply:
If we're using a `Button` but overriding basically all the key aspects of it (style, semantics, behavior), then we should consider not using `Button` at all here. The complexity added for non-buttony use cases is going to make it increasingly difficult to maintain this.

FWIW the Headless UI and React Spectrum Picker examples we've been referencing don't use buttons.

Reply:
I knew you were going to suggest this 😅

Yeah, I'll do the changes. At this point it probably makes sense to use something else